Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2316,9 +2357,23 @@ AttributedString editingAttributedString(const SimpleRange& range, IncludeImages
         auto node = it.node();
 
         if (RefPtr imageElement = dynamicDowncast<HTMLImageElement>(node); imageElement && includeImages == IncludeImages::Yes) {
-            auto fileWrapper = fileWrapperForElement(*imageElement);
-            auto attachment = adoptNS([[NSTextAttachment alloc] initWithFileWrapper:fileWrapper.get()]);
-            [string appendAttributedString:[NSAttributedString attributedStringWithAttachment:attachment.get()]];
+            BOOL addedImage = NO;
+#if ENABLE(MULTI_REPRESENTATION_HEIC)
+            if (imageElement->isMultiRepresentationHEIC()) {
+                if (RefPtr image = imageElement->image()) {
+                    if (WebMultiRepresentationHEICAttachment *attachment = image->adapter().multiRepresentationHEIC()) {
+                        [string appendAttributedString:[NSAttributedString attributedStringWithAttachment:static_cast<NSTextAttachment *>(attachment)]];
+                        addedImage = YES;
+                    }
+                }
+            }
+#endif
+
+            if (!addedImage) {
+                auto fileWrapper = fileWrapperForElement(*imageElement);
+                auto attachment = adoptNS([[NSTextAttachment alloc] initWithFileWrapper:fileWrapper.get()]);
+                [string appendAttributedString:[NSAttributedString attributedStringWithAttachment:attachment.get()]];
+            }
```

Comment:
can this maybe be refactored into its own method with early returns to ease readability?